pkg/splunk/enterprise/clustermaster.go

```diff
@@ -76,23 +76,36 @@ func ApplyClusterMaster(client splcommon.ControllerClient, cr *enterprisev1.Clus
 		return result, err
 	}
 
-	// ToDo sgontla: Handle
-	// 1. Spec update
-	// 2. S3 credentails change
-	// 3. Probe times expired
-	if !reflect.DeepEqual(cr.Status.AppContext.AppFrameworkConfig, cr.Spec.AppFrameworkConfig) {
-		// TBD: Probe the remote store, and handle any changes on remote store
-
-		cr.Status.AppContext.AppFrameworkConfig = cr.Spec.AppFrameworkConfig
-	}
-
 	defer func() {
 		err = client.Status().Update(context.TODO(), cr)
 		if err != nil {
 			scopedLog.Error(err, "Status update failed")
 		}
 	}()
 
+	//check if the apps need to be downloaded from remote storage
+	if cr.Spec.CommonSplunkSpec.Mock != true && !reflect.DeepEqual(cr.Status.AppContext.AppFrameworkConfig, cr.Spec.AppFrameworkConfig) {
```

Comment:
This is only executed when a CRD is applied and the AppFrameworkConfig changes, right?

Reply:
yup